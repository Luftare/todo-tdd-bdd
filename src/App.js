import React, { useState } from 'react';

function App() {
  const [items, setItems] = useState([]);

  const handleClick = () => {
    setItems([...items, 'something']);
  };

  return (
    <div>
      <h1 data-title-e2e-test>Todo app</h1>
      <input data-new-item-text-field-e2e-test />
      <button
        data-create-new-item-button-e2e-test
        data-create-new-item-button-unit-test
        onClick={handleClick}
      >
        Add
      </button>
      <ul data-todo-list-e2e-test data-todo-list-unit-test>
        {items.map(item => (
          <li key={item} data-todo-item-unit-test></li>
        ))}
      </ul>
    </div>
  );
}

export default App;
