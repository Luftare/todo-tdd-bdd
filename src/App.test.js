import React from 'react';
import TestRenderer, { act } from 'react-test-renderer';
import App from './App';

describe('App', () => {
  let renderer;
  let instance;

  beforeEach(() => {
    renderer = TestRenderer.create(<App />);
    instance = renderer.root;
  });

  it('renders correctly', () => {
    const output = renderer.toJSON();

    expect(output).toMatchSnapshot();
  });

  describe('given create new item button is clicked', () => {
    let todoList;
    let item;

    beforeEach(() => {
      todoList = instance.find(e => e.props['data-todo-list-unit-test']);
      const createNewItemButton = instance.find(
        e => e.props['data-create-new-item-button-unit-test']
      );

      act(() => {
        createNewItemButton.props.onClick();
      });

      item = instance.find(e => e.props['data-todo-item-unit-test']);
    });

    it('new item is added to items list', () => {
      expect(item).toBeDefined();
    });
  });
});
