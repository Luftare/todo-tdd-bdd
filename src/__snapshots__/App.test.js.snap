// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <h1
    data-title-e2e-test={true}
  >
    Todo app
  </h1>
  <input
    data-new-item-text-field-e2e-test={true}
  />
  <button
    data-create-new-item-button-e2e-test={true}
    data-create-new-item-button-unit-test={true}
    onClick={[Function]}
  >
    Add
  </button>
  <ul
    data-todo-list-e2e-test={true}
    data-todo-list-unit-test={true}
  />
</div>
`;
